packages/react-composites/src/composites/ChatComposite/adapter/FileSharingAdapter.ts

```diff
@@ -0,0 +1,158 @@
+
+/**
+ * @beta
+ */
+export interface FileUploadAdapter {
+  registerFileUploads?: (fileUploads: ObservableFileUpload[]) => void;
+  clearFileUploads?: () => void;
+  cancelFileUpload?: (id: string) => void;
+}
+
+/**
+ * @internal
+ */
+class FileSharingContext {
+  private chatContext: ChatContext;
+
+  constructor(chatContext: ChatContext) {
+    this.chatContext = chatContext;
+  }
+
+  public getFileUploads(): Record<string, FileUploadState> | undefined {
+    return this.chatContext.getState().fileUploads;
+  }
+
+  public setFileUploads(fileUploads: ObservableFileUpload[]): void {
+    const fileUploadsMap = fileUploads.reduce((map: Record<string, FileUploadState>, fileUpload) => {
+      map[fileUpload.id] = {
+        id: fileUpload.id,
+        filename: fileUpload.file.name,
+        progress: 0
+      };
+      return map;
+    }, {});
+
+    this.chatContext.setState(
+      produce(this.chatContext.getState(), (draft) => {
+        draft.fileUploads = fileUploadsMap;
+      })
+    );
+  }
+
+  public updateFileUpload(
+    id: string,
+    data: Partial<Pick<FileUploadState, 'progress' | 'metadata' | 'errorMessage'>>
+  ): void {
+    this.chatContext.setState(
+      produce(this.chatContext.getState(), (draft: ChatAdapterState) => {
+        if (draft.fileUploads?.[id]) {
+          draft.fileUploads[id] = {
+            ...draft.fileUploads?.[id],
+            ...data
+          };
+        }
+      })
+    );
+  }
+
+  public deleteFileUpload(id: string): void {
+    this.chatContext.setState(
+      produce(this.chatContext.getState(), (draft: ChatAdapterState) => {
+        delete draft?.fileUploads?.[id];
+      })
+    );
+  }
+}
+
+/**
+ * @internal
+ */
+export class FileSharingAdapter implements FileUploadAdapter {
+  private context: FileSharingContext;
+  private fileUploads: ObservableFileUpload[] = [];
+
+  constructor(chatContext: ChatContext) {
+    this.context = new FileSharingContext(chatContext);
+  }
+
+  private findFileUpload(id: string): ObservableFileUpload | undefined {
+    return this.fileUploads.find((fileUpload) => fileUpload.id === id);
+  }
+
+  private deleteFileUpload(id: string): void {
+    this.fileUploads = this.fileUploads.filter((fileUpload) => fileUpload.id !== id);
+  }
+
+  registerFileUploads(fileUploads: ObservableFileUpload[]): void {
+    this.fileUploads = this.fileUploads.concat(fileUploads);
+    this.context.setFileUploads(this.fileUploads);
+    this.fileUploads.forEach(this.subscribeAllEvents);
+  }
+
+  clearFileUploads(): void {
+    this.context.setFileUploads([]);
+    this.fileUploads.forEach(this.unsubscribeAllEvents);
+  }
+
+  cancelFileUpload(id: string): void {
+    this.context.deleteFileUpload(id);
+    const fileUpload = this.findFileUpload(id);
+    if (!fileUpload) {
+      throw new Error('File upload not found');
+    }
+    this.unsubscribeAllEvents(fileUpload);
+    this.deleteFileUpload(id);
+  }
+
+  getFileSharingMetadata(): FileSharingMetadata {
```

Comment:
Why is this needed? This information should be available from the `state`.

Reply:
Right, I am gonna take it out of the adapter and make it a private util function in this file.
Doesn't need to be part of the adapter.